bundles/org.openhab.persistence.jdbc/src/main/java/org/openhab/persistence/jdbc/internal/JdbcPersistenceService.java

```diff
@@ -231,4 +244,12 @@ public void updateConfig(Map<Object, Object> configuration) {
     public List<PersistenceStrategy> getDefaultStrategies() {
         return List.of(PersistenceStrategy.Globals.CHANGE);
     }
+
+    @Override
+    public boolean remove(FilterCriteria filter) throws IllegalArgumentException {
+        if (filter.getItemName() == null) {
+            throw new IllegalArgumentException("Item name must not be null");
+        }
+        throw new UnsupportedOperationException("Not implemented yet");
```

Comment:
Is it a lot of work to implement this? If not implemented I'd always throw the `UnsupportedOperationException` so users will not waste time on adding a missing item name and then eventually run into `UnsupportedOperationException`. :wink:

Reply:
I am not sure. will look into it.

Reply:
@wborn I implemented the missing piece of code and added basic unit tests for the generation of queries.